File: velolabs_repos/lattis_android/app/src/main/java/com/lattis/ellipse/domain/repository/RideRepository.java
package com.lattis.ellipse.domain.repository;

import com.lattis.ellipse.data.network.model.response.history.RideHistoryResponse;
import com.lattis.ellipse.data.network.model.response.ride.RideSummaryResponse;
import com.lattis.ellipse.data.network.model.response.ride.UpdateTripResponse;
import com.lattis.ellipse.domain.model.Bike;
import com.lattis.ellipse.domain.model.Location;
import com.lattis.ellipse.domain.model.Ride;

import io.reactivex.Observable;

public interface RideRepository {

    Observable<Ride> startRide(Bike bike, Location location, boolean first_lock_connect);
    Observable<Boolean> endRide(int trip_id, Location location, int parkingId, String imageURL,boolean isReportDamage, Integer lock_battery, Integer bike_battery);
    Observable<UpdateTripResponse> updateRide(int trip_id, double[][] steps);
    Observable<Ride> getRide();
    Observable<RideSummaryResponse> getRideSummary(int trip_id);
    Observable<Boolean> rateRide(int trip_id, int rating);
    Observable<Ride> saveRide(Ride ride);
    Observable<Boolean> deleteRide();
    Observable<RideHistoryResponse> getRideHistory();
}

File: velolabs_repos/lattis_android_v2/domain/src/main/java/com/lattis/domain/repository/RideRepository.kt
package com.lattis.domain.repository

import com.lattis.domain.models.RideSummary
import com.lattis.domain.models.UpdateTripData
import com.lattis.domain.models.Location
import com.lattis.domain.models.Ride
import com.lattis.domain.models.RideHistory
import io.reactivex.rxjava3.core.Observable

interface RideRepository {
    fun getRide(): Observable<Ride>
    fun saveRide(ride: Ride): Observable<Ride>
    fun deleteRide(): Observable<Boolean>
    fun getRideSummary(trip_id: Int): Observable<RideSummary>
    fun updateRide(trip_id: Int, steps: Array<DoubleArray>): Observable<UpdateTripData>
    fun startRide(ride: Ride, location: Location, first_lock_connect: Boolean,device_token:String): Observable<Ride>
    fun endRide(
        trip_id: Int,
        location: Location?,
        parkingId: Int,
        imageURL: String?,
        isReportDamage: Boolean,
        lock_battery: Int?,
        bike_battery: Int?
    ): Observable<RideSummary>


    fun rateRide(trip_id: Int, rating: Int): Observable<Boolean>

    fun getRideHistory(): Observable<RideHistory>
}